extensions/parquet/base/src/main/java/io/deephaven/parquet/base/AbstractBulkValuesWriter.java

```diff
@@ -41,6 +41,17 @@ public int writeBulkVector(@NotNull final BUFFER_TYPE bulkValues,
         return applyDlAndRl(vectorSizes, rlEncoder, dlEncoder, nullsOffsets);
     }
 
+    /**
+     * This method is used to provide a specialized definition of NULL for PlainIntChunkedWriter. For other writers,
+     * this method does nothing.
+     *
+     * @param nullDefinition The value which will be interpreted as NULL for this writer
+     */
+    @Override
+    public void setNull(int nullDefinition) {
+        throw new UnsupportedOperationException("This method should not be called");
```

Comment:
This doesn't look like "doing nothing". Maybe you'd prefer to leave this method abstract? Also, no need to replicate the JavaDoc at every layer, unless you want to be more specific.

Reply:
I was a bit confused here. So all writers inherit from AbstractBulkValuesWriter class which inherits from the abstract class BulkWriter. 
I wanted to give a specialized definition of this method for PlainIntChunkedWriter and raise an exception for all other chunkedWriter types. So I provided a default implementation in this class, so that the other chunkedWriter types don't have to implement it. Do you think there is a better way to implement this?

Reply:
I like the solution you've arrived at. This is better done at construction time.

In a hypothetical world where we didn't do that, see my other comment about making this take `Object` and thus being generic.

At that point, we'd have a decision to make:
1. add the null-value override functionality to all writers, or
1. throw a UOE as you've done for writers that don't support that, or
1. have `setNull` be a no-op for other writers.

The JavaDoc implies (to me) that you've taken approach 3, but you've actually taken approach 2.

Reply:
The inconsistency between the JavaDoc and the implementation was my mistake, thanks for pointing that out.
Even though we are not doing this now, I like the approach 1 where the functionality is more generic.